sycl/test/separate-compile/cpp_version_mismatch_test_1.cpp

```diff
@@ -0,0 +1,56 @@
+// >> ---- device compilation
+// RUN: %clangxx -std=c++14 -fsycl-device-only -Xclang -fsycl-int-header=sycl_ihdr_a.h %s -c -I %sycl_include
+//
+// >> ---- host compilation: cpp version mismatch
+// RUN: not %clangxx -std=c++11 -include sycl_ihdr_a.h -c %s -I %sycl_include 2>&1 | FileCheck %s 
+// CHECK: C++ version for host compilation does not match C++ version used for device compilation

```

Comment:
I think you can make this test as FE test. Please move it to clang/test/SemaSYCL .

Reply:
Capturing llvm-lit test folder transfer into a separate issue.